### PR TITLE
Upgrade k8s dependencies 1.30 , golang 1.22, CSI Sidecar versions

### PR DESCRIPTION
**Pull request by sameshai**
**Changes for the PR**

- Upgrade k8s dependencies from 1.28 to 1.30
- Upgrade GoLang from 1.20 to 1.22
- Upgrade CSI side car versions 
    - csi-provisioner:v5.0.2
    - csi-resizer:v1.11.2
    - livenessprobe:v2.13.1
    - csi-node-driver-registrar:v2.11.1

### CODE REVIEW

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [74.38%]

**Review by arahamad — DISMISSED** [review]
lgtm
please verify and compatibility with cluster versions

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [74.14%]

File: .travis.yml [hunk 1]
```diff
@@ -16,11 +15,6 @@ cache:
 
 sudo: true
 
-addons:
-  apt:
-    packages:
-      - docker-ce
```
Comment (skorati) [inline .travis.yml:22]: Is this not needed now?

File: pkg/ibmcsidriver/controller.go [hunk 2]
```diff
@@ -591,3 +591,9 @@ func (csiCS *CSIControllerServer) ControllerGetVolume(ctx context.Context, req *
 	_ = context.WithValue(ctx, provider.RequestID, requestID)
 	return nil, commonError.GetCSIError(ctxLogger, commonError.MethodUnimplemented, requestID, nil, "ControllerGetVolume")
 }
+
+// ControllerModifyVolume ...
+func (csiCS *CSIControllerServer) ControllerModifyVolume(ctx context.Context, req *csi.ControllerModifyVolumeRequest) (*csi.ControllerModifyVolumeResponse, error) {
```
Comment (skorati) [inline pkg/ibmcsidriver/controller.go:596]: For calling `ControllerModifyVolume` method , do we need `patch` permissions?

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [74.38%]

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [74.38%]

File: .travis.yml (hunk 1 above)
Comment (sameshai) [reply .travis.yml:22]: Yes there is some build issue earlier we had introduced this I have upgraded the golang version and travis to latest

File: pkg/ibmcsidriver/controller.go (hunk 2 above)
Comment (sameshai) [reply pkg/ibmcsidriver/controller.go:596]: The patch permission is just for the HonourPVReclaim Policy feature
https://github.ibm.com/alchemy-containers/armada-storage/issues/6528
https://github.com/kubernetes-csi/external-provisioner/pull/1155

**Review by skorati — APPROVED** [review]

**Review by prankulmahajan — APPROVED** [review]
lgtm

**Pull request merged**